R/dtree_cross_validation.R

```diff
@@ -0,0 +1,531 @@
+    actualPosIndic <- which(actual == posClass)
+    nActualPos <- length(actualPosIndic)
+    nCorrectPos <- sum(scoredOutput[actualPosIndic] == posClass)
+    nPredPos <- sum(scoredOutput == posClass)
+    precision <- nCorrectPos/nPredPos
+    recall <- nCorrectPos/nActualPos
+    F1 <- 2*(precision*recall)/(precision + recall)
+    AUC <- performance(prediction.obj = predictionObj, measure = "auc")
+    AUC <- unlist(AUC@y.values)
+    percentClass1Right <- sum(scoredOutput[which(actual == (extras$levels)[1])] == (extras$levels)[[1]])/length(which(actual == (extras$levels)[1]))
+    percentClass2Right <- sum(scoredOutput[which(actual == (extras$levels)[2])] == (extras$levels)[[2]])/length(which(actual == (extras$levels)[2]))
+    outVec <- c(mid = modelIndic, trial = trialIndic, fold = foldIndic, Accuracy_Overall = overallAcc, Accuracy_Class_1 = percentClass1Right, Accuracy_Class_2 = percentClass2Right, F1 = F1, AUC = AUC)
+  } else {
+    #Compute accuracy by class
+    outVec <- vector(length = length((extras$levels)))
+    for (l in 1:length((extras$levels))) {
+      tempPred <- scoredOutput[actual == (extras$levels)[[l]]]
+      nCorrect <- sum(tempPred == (extras$levels)[[l]])
+      thisAcc <- nCorrect/sum(actual == (extras$levels)[[l]])
+      outVec[l] <- thisAcc
+      names(outVec)[l] <- paste0("Accuracy_Class_", l)
+    }
+    outVec <- c(mid = modelIndic, trial = trialIndic, fold = foldIndic, Accuracy_Overall = overallAcc, outVec)
+  }
+  return(outVec)
+}
+
+# Functions to Generate Output
+#' @import reshape2
+generateConfusionMatrices <- function(outData, extras) {
+  outvec <- vector(length = length(extras$levels))
+  pasteClass <- function(nameOfClass) {
+    paste0("Class_", nameOfClass)
+  }
+  names(outvec) <- sapply(X = (extras$levels), FUN = pasteClass, simplify = TRUE)
+  for (i in 1:length(extras$levels)) {
+    outvec[i] <- length(which((outData$actual) == ((extras$levels)[i])))
+  }
+  return(c(mid = unique(outData$mid), trial = unique(outData$trial), fold = unique(outData$fold), Predicted_class = as.character(unique(outData$response)), outvec))
+}
+
+generateOutput3 <- function(data, extras, modelNames) {
+  d <- plyr::ddply(data, c("trial", "fold", "mid", "response"), generateConfusionMatrices,
+             extras = extras
+  )
+  d$Model <- modelNames[as.numeric(d$mid)]
+  d$Type <- rep.int('Classification', times = length(d$Model))
+  d <- subset(d, select = -c(mid, response))
+  d <- reshape2::melt(d, id = c('trial', 'fold', 'Model', 'Type', 'Predicted_class'))
+  colnames(d) <- c('Trial', 'Fold', 'Model', 'Type', 'Predicted_class', 'Variable', 'Value')
+  return(d)
+}
+
+generateOutput2 <- function(data, extras, modelNames) {
+  fun <- if (is.null(extras$levels)) {
+    getMeasuresRegression
+  } else {
+    getMeasuresClassification
+  }
+  d <- plyr::ddply(data, c("trial", "fold", "mid"), fun, extras = extras)
+  d$Model <- modelNames[as.numeric(d$mid)]
+  d <- subset(d, select = -c(mid))
```

Comment:
Change this to `d <- d[,-d$mid]`. `subset` is good for interactive use, not while writing packages. This change will fix the warning on Line 924 on the Travis log.